src/Storages/VirtualColumnUtils.h

```diff
@@ -33,6 +33,7 @@ bool prepareFilterBlockWithQuery(const ASTPtr & query, ContextPtr context, Block
 /// Only elements of the outer conjunction are considered, depending only on the columns present in the block.
 /// If `expression_ast` is passed, use it to filter block.
 void filterBlockWithQuery(const ASTPtr & query, Block & block, ContextPtr context, ASTPtr expression_ast = {});
+void filterBlockWithQuery(ActionsDAGPtr dag, Block & block, ContextPtr context);
```

Comment:
Probably it's better to add a comment, that this overload should be used in the new infrastructure.